File: AD-EYE/ROS_Packages/src/AD-EYE/src/publish_detected_objects.py
#!/usr/bin/env python

import rospy
from visualization_msgs.msg import Marker
from visualization_msgs.msg import MarkerArray
from geometry_msgs.msg import Pose
import sys

p = Pose()
p.position.x = 84.599590301513672
p.position.y = 171.27021789550781
p.position.z = -2.22
p.orientation.x = 0.0
p.orientation.y = 0.0
p.orientation.z = 0.0
p.orientation.w = 0.0

marker_msg = Marker()
marker_array_msg = MarkerArray()
marker_msg.header.frame_id= "world"
marker_msg.type = Marker.CUBE
marker_msg.action = Marker.ADD
marker_msg.pose = p

marker_msg.scale.x = 5
marker_msg.scale.y = 5
marker_msg.scale.z = 5

marker_msg.color.r = 255
marker_msg.color.g = 1
marker_msg.color.b = 1
marker_msg.color.a = 0.5

marker_array_msg.markers.append(marker_msg)

if __name__ == '__main__':
    rospy.init_node('test_box_publisher', anonymous=True)
    rate = rospy.Rate(10.0)
    pub = rospy.Publisher('/objects_markers', MarkerArray, queue_size=1)
    counter = 0
    while not rospy.is_shutdown():
        markerx = marker_array_msg.markers[0].pose.position.x
        if counter%10 == 0:
            if(markerx > 200):
                markerx = 84.59
            markerx += 10
        marker_array_msg.markers[0].pose.position.x = markerx
        pub.publish(marker_array_msg)
        counter +=1
        rate.sleep()

File: AD-EYE/ROS_Packages/src/AD-EYE/src/publish_detected_objects.cpp
#include <ros/ros.h>
#include <visualization_msgs/MarkerArray.h>
#include <visualization_msgs/Marker.h>
#include <std_msgs/ColorRGBA.h>

class DetectedObjectsPublisher
{
    private:
        ros::NodeHandle nh_;
        ros::Publisher pub_markers_;
        std::string ros_namespace_;

        std::string markers_out_topic = "/objects_markers";
        float marker_display_duration_ = 0.2;

        /*!
        * \brief function to create color of type ROS  std_msgs::ColorRGBA from RGBA values
        * \param in_color vector of size 4  containing RGBA values of color 
        */
        static std_msgs::ColorRGBA parseColor(const std::vector<double> in_color)
        {
            std_msgs::ColorRGBA color;
            float r,g,b,a;
            if (in_color.size() == 4) //r,g,b,a
            {
                color.r = in_color[0]/255.f;
                color.g = in_color[1]/255.f;
                color.b = in_color[2]/255.f;
                color.a = in_color[3];
            }
            return color;
        }
    public:
        DetectedObjectsPublisher(ros::NodeHandle& nh): nh_(nh)
        {
            pub_markers_ = nh_.advertise<visualization_msgs::MarkerArray>(
                                markers_out_topic, 1);
            ros_namespace_ = ros::this_node::getNamespace();
        }
        void run()
        {
            visualization_msgs::Marker marker;

            marker.lifetime = ros::Duration(marker_display_duration_);
            marker.header.frame_id= "world";
            marker.header.stamp = ros::Time::now();
            marker.type = visualization_msgs::Marker::CUBE;
            marker.action = visualization_msgs::Marker::ADD;
            marker.ns = ros_namespace_ + "/detected_object_markers";

            marker.pose.position.x = 84.59;
            marker.pose.position.y = 175.01;
            marker.pose.position.z = -2.22;
            marker.pose.orientation.x = 0.0;
            marker.pose.orientation.y = 0.0;
            marker.pose.orientation.z = 0.0;
            marker.pose.orientation.w = 0.0;
            marker.scale.x = 3;
            marker.scale.y = 3;
            marker.scale.z = 3;
            marker.color = parseColor({255, 0, 0, 1.0}); // color:red
            
            ros::Rate loop_rate(20);
            int count = 0;
            float markerx;
            while (ros::ok())
            {

                visualization_msgs::MarkerArray markers;
                markerx = marker.pose.position.x;
                if (count%10 == 0)
                {
                    if(markerx > 200)
                        markerx = 84.59;
                    markerx += 10;
                }
                    
                marker.pose.position.x = markerx;
                markers.markers.push_back(marker);
                pub_markers_.publish(markers);

                ros::spinOnce();

                loop_rate.sleep();
                ++count;
            }

        }

};

int main(int argc, char** argv)
{
    ros::init(argc, argv, "DetectedObjectsPublisher");
    ros::NodeHandle nh;
    DetectedObjectsPublisher dop(nh);
    dop.run();
    return 0;
}